common/src/main/java/org/wso2/testgrid/common/config/ScenarioConfig.java

```diff
@@ -107,5 +160,61 @@ public String getRemoteBranch() {
     public void setRemoteBranch(String remoteBranch) {
         this.remoteBranch = remoteBranch;
     }
+
+    /**
+     * Returns the list of input parameters required for the test script.
+     *
+     * @return List of inputParameters
+     */
+    public Map<String, String> getInputParameters() {
+        return inputParameters;
+    }
+
+    /**
+     * Sets the input parameters required to run test script.
+     *
+     * @param inputParameters List of input parameters taken from scenarioConfig
+     */
+    public void setInputParameters(Map<String, String> inputParameters) {
+        this.inputParameters = inputParameters;
+    }
+
+    /**
+     * Returns the file name of the test script (e.g. test.sh).
+     *
+     * @return file name
+     */
+    public String getFile() {
+        return file;
+    }
+
+    /**
+     * Sets the filename of the test script.
+     *
+     * @param file file name
+     */
+    public void setFile(String file) {
+        this.file = file;
+    }
+
+    public String getOutputDir() {
+        if (StringUtil.isStringNullOrEmpty(outputDir)) {
+            try {
+                URL url = new URL(remoteRepository);
+                String [] splitbyDash = url.getPath().split("/");
+                String [] splitByDot = splitbyDash[splitbyDash.length - 1].split("\\.");
```

Comment:
This is an unusual logic, and hence not very portable probably. What assumptions did you make here?
For instance, did you assume the remoteRepository will always be a 'https' url?